Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,587 @@
+    ((differentiable_id.add_const 1).div_const 2)).div_const 2
+
+/-!
+## Parity and functional equations
+-/
+
+lemma completedHurwitzZetaOdd_neg (a : UnitAddCircle) (s : ℂ) :
+    completedHurwitzZetaOdd (-a) s = -completedHurwitzZetaOdd a s := by
+  simp only [completedHurwitzZetaOdd, StrongFEPair.Λ, hurwitzOddFEPair, mellin, Function.comp_def,
+    oddKernel_neg, ofReal_neg, smul_neg]
+  rw [integral_neg, neg_div]
+
+lemma completedSinZeta_neg (a : UnitAddCircle) (s : ℂ) :
+    completedSinZeta (-a) s = -completedSinZeta a s := by
+  simp only [completedSinZeta, StrongFEPair.Λ, mellin, StrongFEPair.symm, WeakFEPair.symm,
+    hurwitzOddFEPair, Function.comp_def, sinKernel_neg, ofReal_neg, smul_neg]
+  rw [integral_neg, neg_div]
+
+
+/-- Functional equation for the odd Hurwitz zeta function. -/
+theorem completedHurwitzZetaOdd_one_sub (a : UnitAddCircle) (s : ℂ) :
+    completedHurwitzZetaOdd a (1 - s) = completedSinZeta a s := by
+  rw [completedHurwitzZetaOdd, completedSinZeta,
+    (by { push_cast; ring } : (1 - s + 1) / 2 = ↑(3 / 2 : ℝ) - (s + 1) / 2),
+    ← hurwitzOddFEPair_k, (hurwitzOddFEPair a).functional_equation ((s + 1) / 2),
+    hurwitzOddFEPair_ε, one_smul]
+
+/-- Functional equation for the odd Hurwitz zeta function (alternative form). -/
+lemma completedSinZeta_one_sub (a : UnitAddCircle) (s : ℂ) :
+    completedSinZeta a (1 - s) = completedHurwitzZetaOdd a s := by
+  rw [← completedHurwitzZetaOdd_one_sub, sub_sub_cancel]
+
+/-!
+## Relation to the Dirichlet series for `1 < re s`
+-/
+
+/-- Formula for `completedSinZeta` as a Dirichlet series in the convergence range
+(first version, with sum over `ℤ`). -/
+lemma hasSum_int_completedSinZeta (a : ℝ) {s : ℂ} (hs : 1 < re s) :
+    HasSum (fun n : ℤ ↦ Gammaℝ (s + 1) * (-I) * Int.sign n *
+    cexp (2 * π * I * a * n) / (↑|n| : ℂ) ^ s / 2) (completedSinZeta a s) := by
+  let c (n : ℤ) : ℂ := -I * cexp (2 * π * I * a * n) / 2
+  have hc (n : ℤ) : ‖c n‖ = 1 / 2 := by
+    simp_rw [c, (by { push_cast; ring } : 2 * π * I * a * n = ↑(2 * π * a * n) * I), norm_div,
+      RCLike.norm_ofNat, norm_mul, norm_neg, norm_I, one_mul, norm_exp_ofReal_mul_I]
+  have hF t (ht : 0 < t) :
+      HasSum (fun n ↦ c n * n * rexp (-π * n ^ 2 * t)) (sinKernel a t / 2) := by
+    refine ((hasSum_int_sinKernel a ht).div_const 2).congr_fun fun n ↦ ?_
+    rw [div_mul_eq_mul_div, div_mul_eq_mul_div, mul_assoc (-I), mul_assoc (-I),
+      mul_assoc (-I), mul_assoc (-I), mul_comm (cexp _)]
+  have h_sum : Summable fun i ↦ ‖c i‖ / |↑i| ^ s.re := by
+    simp_rw [hc, div_right_comm]
+    apply Summable.div_const
+    apply Summable.of_nat_of_neg <;>
+    · simp only [Int.cast_neg, abs_neg, Int.cast_natCast, Nat.abs_cast]
+      rwa [summable_one_div_nat_rpow]
+  have := mellin_div_const .. ▸ hasSum_mellin_pi_mul_sq' (zero_lt_one.trans hs) hF h_sum
+  refine this.congr_fun fun n ↦ ?_
+  rw [← Int.cast_abs, ofReal_intCast]
+  have : (Int.sign n : ℂ) = SignType.sign (n : ℝ) := by
```

Comment:
This should be much simpler. It looks like API for `sign` is missing...

Reply:
```lean
  have : (Int.sign n : ℂ) = SignType.sign (n : ℝ) := by
    simp only [Int.sign_eq_sign, sign_apply]
    norm_cast
    split_ifs
    · simp only [SignType.coe_one, Int.cast_one]
    · simp only [SignType.coe_neg_one, Int.cast_neg, Int.reduceNeg, Int.cast_one]
    · simp only [SignType.coe_zero, Int.cast_zero]
```
looks more principled (and is faster, though still slower than I think it should be).

At a cost of 600 hb, one can shorten the end game to
```lean
    split_ifs <;> simp
```